src/topology/algebra/ordered.lean

```diff
@@ -1295,12 +1315,89 @@ begin
   exact tendsto_le_left inf_le_left (Mf.tendsto _)
 end
 
-/-- A continuous monotone function sends indexed infimum to indexed infimum in conditionally complete
-lattices, under a boundedness assumption. -/
+/-- A continuous monotone function sends indexed infimum to indexed infimum in conditionally
+complete lattices, under a boundedness assumption. -/
 lemma cinfi_of_cinfi_of_monotone_of_continuous {f : α → β} {g : γ → α}
   (Mf : continuous f) (Cf : monotone f) (H : bdd_below (range g)) : f (infi g) = infi (f ∘ g) :=
 by rw [infi, cInf_of_cInf_of_monotone_of_continuous Mf Cf (range_nonempty _) H, ← range_comp, infi]
 
+/-- A bounded connected subset of a conditionally complete lattice includes the open interval
+`(Inf s, Sup s)`. -/
+lemma is_connected.Ioo_cInf_cSup_subset {s : set α} (hs : is_connected s) (hb : bdd_below s)
+  (ha : bdd_above s) :
+  Ioo (Inf s) (Sup s) ⊆ s :=
+λ x hx, let ⟨y, ys, hy⟩ := (is_glb_lt_iff (is_glb_cInf hs.nonempty hb)).1 hx.1 in
+let ⟨z, zs, hz⟩ := (lt_is_lub_iff (is_lub_cSup hs.nonempty ha)).1 hx.2 in
+hs.is_preconnected.Icc_subset ys zs ⟨le_of_lt hy, le_of_lt hz⟩
+
+lemma is_connected.Ioi_cInf_subset {s : set α} (hs : is_connected s) (hb : bdd_below s)
```

Comment:
For this one and the next one, you only need `is_preconnected s`, as the empty set is bounded above (as the space is not empty since it contains `Inf \emptyset`).